ListDetail/src/main/java/com/microsoft/device/display/samples/listdetail/ui/view/MainPage.kt

```diff
@@ -7,19 +7,25 @@ package com.microsoft.device.display.samples.listdetail.ui.view
 
 import androidx.compose.foundation.Image
 import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.res.painterResource
-import com.microsoft.device.display.samples.listdetail.models.AppStateViewModel
 import com.microsoft.device.dualscreen.twopanelayout.TwoPaneLayout
 import com.microsoft.device.dualscreen.twopanelayout.TwoPaneMode
 import com.microsoft.device.dualscreen.windowstate.WindowState
 
 @Composable
-fun ListDetailApp(viewModel: AppStateViewModel, windowState: WindowState) {
+fun ListDetailApp(windowState: WindowState) {
+    var selectedIndex by rememberSaveable { mutableStateOf(0) }
+    val updateSelectedIndex: (Int) -> Unit = { newIndex -> selectedIndex = newIndex }
+
     TwoPaneLayout(
         paneMode = TwoPaneMode.HorizontalSingle,
-        pane1 = { ListViewWithTopBar(viewModel) },
-        pane2 = { DetailViewWithTopBar(windowState.isDualPortrait(), viewModel) }
+        pane1 = { ListViewWithTopBar(windowState.isDualPortrait(), selectedIndex, updateSelectedIndex) },
+        pane2 = { DetailViewWithTopBar(windowState.isDualPortrait(), selectedIndex) }
```

Comment:
Another suggestion - to make testing easier, you could add a "stateless" version of the app composable so you don't have to create a `WindowState` object for each test:

```@Composable
fun ListDetailApp(windowState: WindowState) {
    var selectedIndex by rememberSaveable { mutableStateOf(0) }
    val updateSelectedIndex: (Int) -> Unit = { newIndex -> selectedIndex = newIndex }

    ListDetailAppContent(windowState.isDualPortrait(), selectedIndex, updateSelectedIndex)
}

@Composable
fun ListDetailAppContent(isDualScreen: Boolean, selectedIndex: Int, updateSelectedIndex: (Int) -> Unit) {
    TwoPaneLayout(
        paneMode = TwoPaneMode.HorizontalSingle,
        pane1 = { ListViewWithTopBar(isDualScreen, selectedIndex, updateSelectedIndex) },
        pane2 = { DetailViewWithTopBar(isDualScreen, selectedIndex) }
    )
}
```

Reply:
I think that it is better to pass `WindowState` into each test case, since I am testing three different states here. 
```
ListDetailApp(WindowState(hasFold = true, isFoldHorizontal = false))
ListDetailApp(WindowState(hasFold = true, isFoldHorizontal = true))
ListDetailApp(WindowState(hasFold = false, isFoldHorizontal = false))
```

Reply:
Sounds good!